server/socketServer.ts

```diff
@@ -26,18 +26,27 @@ const io = new Server(socketConfig);
 // and remembers the original insertion order of the keys
 // const socketToPlayerId = new Map<number, string>();
 const playerData = new Map<string, PlayerData>();
-// const playerData: [Player] = [];
+const playerClients: string[] = [];
```

Comment:
Moved the client list to the server - this is the best and most optimal way to maintain a source of truth. Since a client could disconnect at any time, and we can't emit to the server from the FE when a client just disconnects (at least I believe so), we can easily adjust the client list from the server and emit the updated list right back (see the `socket.on('disconnect')` below). This also allows us to easily update which player has which player number/index.